### ISSUE TITLE
Sign in with Google module not in alphabetical order in list of Site Kit modules

### ISSUE THREAD

**Issue by adamdunnage** [google/site-kit-wp#9877]
## Bug Description

The Sign in With Google module is not listed in alphabetical order in the list of Site Kit connected modules. It appears last in the order after Tag Manager.

## Steps to reproduce

* Ensure `signInWithGoogleModule` feature flag is enabled
* Connect all Site Kit modules
* View list of Site Kit modules in Site Kit settings

## Screenshots

![Image](https://github.com/user-attachments/assets/b2f66b7e-3f24-44d7-8870-bab300962e15)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The Sign in with Google module should be in alphabetical order (eg. before Tag Manager) in the list of modules, which are otherwise in alphabetical order (except for Search Console, which always comes first).

## Implementation Brief

_Currently we have `order` property defined in `setup_info` method of each module class, but as the number of modules is growing, this is becoming more complicated to update manually and re-order modules with each new inclusion. So I am proposing more automated approach here_
* [ ] in `assets/js/components/settings/SettingsActiveModules.js`, `assets/js/components/settings/SettingsInactiveModules.js` and `assets/js/googlesitekit/modules/datastore/modules.js`
  * Within the current sorting of the modules `.sort( ( a, b ) => a.order - b.order )`
    * Update sorting logic to include alphabetical sorting when when `order` is equal in both items: `a.order - b.order || a.name.localeCompare( b.name )`
* [ ] Remove `order` array key in `setup_info` method from each module class, except from Search Console, which should keep existing `order => 1` so it is always surfacing first in the list. Other modules will inherit the default `order` of `10` from the core modules class

### Test Coverage

* Update any failing VRT
* `assets/js/googlesitekit/modules/datastore/__fixtures__/list.json` will need an update for `order` prop, it should be `10` except for search console, which should remain with order of 1
* `assets/js/googlesitekit/modules/datastore/modules.test.js` might need few tweaks to accommodate new order change, by taking alphabetical order into the account as well

## QA Brief

* With different module feature flags enabled, check the "Connected Services" tab and the "Connect more services" tab within Site Kit settings. All modules in these two tabs should be ordered alphabetically, with Search Console at the top as an exception.

## Changelog entry

* Improve sort order in module list.

**Comment by tofumatt** [comment]
I dig this solution, very nice 👍🏻 

IB ✅

**Comment by jimmymadon** [comment]
@tofumatt @zutigrm 

I just feel that sorting the modules alphabetically in addition to using the order is a bit of an overkill here.
1. We don't frequently add/remove modules to Site Kit, so the `order` property can be updated easily. It will just be a few lines of change to do this 'manually'.
2. Because of Search Console, we cannot completely get rid of the order property anyways. So now having two different ways of calculating the order can make things slightly complicated.
3. There is also a high chance @marrrmarrr would want to re-order this list as everything launches. Perhaps, it might make sense for Analytics to be second above the rest as it is more frequently used? 

So in summary, I feel we can either simply rely on the order property or order all the modules in alphabetical order. The former is ideal since we already have the infrastructure in place.

What do you think? Im still happy to proceed with the IB if we think that is the best way forward.

**Comment by tofumatt** [comment]
> We don't frequently add/remove modules to Site Kit, so the order property can be updated easily. It will just be a few lines of change to do this 'manually'.

True, but we failed to do it in the past, hence this issue, so having it be done automatically makes sense.

> Because of Search Console, we cannot completely get rid of the order property anyways. So now having two different ways of calculating the order can make things slightly complicated.

Right, that's why we have the manual override. The code itself is more complex, but I don't think it's complicated. Additionally, it removes any manual involvement when adding/removing modules in the future.

> There is also a high chance @marrrmarrr would want to re-order this list as everything launches. Perhaps, it might make sense for Analytics to be second above the rest as it is more frequently used?

Where was this discussed? If the plan is to have a very specific order we can do that an opt not to alphabetise things, but if anything I'd like to ask about sorting _everything_ alphabetically eventually, as it would be even more consistent and usable; Search Console itself is not a module whose settings demand much/any interaction, so moving it lower would make more sense.

------

I think it's best to stick with the IB for now, unless there's a point about reordering from point 3 I'm not aware of 🙂
